### ISSUE TITLE
Unable to connect with profile credentials

### ISSUE THREAD

**Issue by PanosBougas** [aws/aws-toolkit-vscode#6230]
## Problem

Cannot connect to AWS using my profile after latest patch.

## Steps to reproduce the issue

1. Go to Command Palette and hit 'Connect to AWS'
2. Use a profile to connect
3. See error 'Unable to authenticate connection'
4. Output:
```
2024-12-13 14:07:08.003 [info] auth: Updating connection state of profile:dev to authenticating
2024-12-13 14:07:08.005 [info] auth: Handling validation error of connection: profile:dev
2024-12-13 14:07:08.005 [info] auth: Updating connection state of profile:dev to invalid
2024-12-13 14:07:08.005 [error] _aws.toolkit.auth.reauthenticate: Error: Unable to authenticate connection
	 -> Error: SSO cache data unexpectedly missing props: ["refreshToken"]
```

The previous Toolkit version didn't have that issue.

-   OS: Windows 11 23H2
-   Visual Studio Code version: 1.95.3
-   AWS Toolkit version: 3.39.0

**Comment by nkomonen-amazon** [comment]
Thanks for reporting.

Fix is here: https://github.com/aws/aws-toolkit-vscode/pull/6234
We will look to release for Tuesday next week.

**A temporary workaround should be to sign out/in and things should start working. If not please downgrade to 3.38.0 until we release this fix next week.**

---

Are you familiar with what your SSO session length should be? 1 day, 30 days, 60 days, ...
It is odd to me that the refreshToken to refresh the session is not present

**Comment by PanosBougas** [comment]
I tried downgrading to 3.38.0 then I was able to connect. Then I tried going back to 3.39.0 assuming that the refreshToken I got from 3.38.0 would work but again I couldn't connect. My session length is 8 hours.

**Comment by justinmk3** [comment]
Can you try the prerelease from https://github.com/aws/aws-toolkit-vscode/releases/tag/prerelease ?

**Comment by PanosBougas** [comment]
yes I tried the prerelease and I got connected!

**Comment by tomcat323** [comment]
This request was addressed in [AWS Toolkit 3.40.0](https://marketplace.visualstudio.com/items?itemName=AmazonWebServices.aws-toolkit-vscode) and/or [Amazon Q 1.41.0](https://marketplace.visualstudio.com/items?itemName=AmazonWebServices.amazon-q-vscode)!

AWS Toolkit changelog: https://github.com/aws/aws-toolkit-vscode/blob/master/packages/toolkit/CHANGELOG.md
Amazon Q changelog: https://github.com/aws/aws-toolkit-vscode/blob/master/packages/amazonq/CHANGELOG.md
Release artifacts: https://github.com/aws/aws-toolkit-vscode/releases
